pkgs/development/python-modules/berkeleydb/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, pkgs
+, python
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "berkeleydb";
+  version = "18.1.8";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-4YMaeQv9hVdA5cEvlS8Z9kbCaYBs/DYnda/Zh4zzJVc=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  buildInputs = [ pkgs.db ];
+
+  checkPhase = ''
+    ${python.interpreter} test.py
+  '';
+
+  # Path to database need to be set.
+  # Somehow the setup.py flag is not propagated.
+  #setupPyBuildFlags = [ "--berkeley-db=${pkgs.db}" ];
+  # We can also use a variable
+  preConfigure = ''
+    export BERKELEYDB_DIR=${pkgs.db.dev};
+  '';
```

Comment:
Sadly, `setupPyBuildFlags` doesn't work when using `pyproject = true`

Instead of doing `export SOMEVAR=value` inside a phase you can add `env.SOMEVAR = value;` into the derivation to achieve the same effect.

Also, I think you should set `BERKELEYDB_INCDIR` instead of just `BERKELEY_DIR`, becuase `_DIR` also expects the library files to be in the directory. The only reason it works is becuase `db` is in `buildInputs` too.

---

So, I think you should either keep `buildInputs` and do
```
env.BERKELEYDB_INCDIR = "${lib.getDev pkgs.db}/include";
```
or remove `buildInputs` and do
```
env = {
  BERKELEYDB_INCDIR = "${lib.getDev pkgs.db}/include";
  BERKELEYDB_LIBDIR = "${lib.getLib pkgs.db}/lib";
};
```